src/com/cft/contactmerge/io/tests/SplitStringUtilitiesTest.java

```diff
@@ -3,16 +3,196 @@
+        String testString = String.join("\t", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitTsvString(testString);
+
+        assertEquals(4, actualParts.size(), createVerificationMessage("result size"));
+
+        for(int i=0; i < actualParts.size(); i++) {
+            assertEquals(expectedParts.get(i), actualParts.get(i), String.format("result[%d]", i));
+        }
+    }
+
+    @Test
+    void splitTsvString_LeadingTrailingSpacesTrimmed() {
+        List<String> expectedParts = Arrays.asList("   ", " Smith  ");
+        String testString = String.join("\t", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitTsvString(testString);
+
+        assertEquals(2, actualParts.size(), createVerificationMessage("result size"));
+
+        assertEquals("", actualParts.get(0), "result[0]");
+        assertEquals("Smith", actualParts.get(1), "result[1]");
+    }
+
+    /* ----------------------------------------------------------
+     * CSV Tests
+     * ---------------------------------------------------------- */
+    @Test
+    void splitCsvString_EmptyString() {
+        List<String> parts = SplitStringUtilities.splitCsvString("");
+
+        assertEquals(1, parts.size(), createVerificationMessage("result size"));
+        assertEquals( "", parts.iterator().next(), createVerificationMessage("first result"));
+    }
+
+    @Test
+    void splitCsvString_SingleValue() {
+        List<String> parts = SplitStringUtilities.splitCsvString("Joe");
+
+        assertEquals(1, parts.size(), createVerificationMessage("result size"));
+        assertEquals( "Joe", parts.iterator().next(), "first result");
+    }
+
+    @Test
+    void splitCsvString_MultipleValues() {
+        List<String> expectedParts = Arrays.asList("Joe", "Smith", "123 Main St");
+        String testString = String.join(",", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitCsvString(testString);
+
+        assertEquals(3, actualParts.size(), createVerificationMessage("result size"));
+
+        for(int i=0; i < actualParts.size(); i++) {
+            assertEquals(expectedParts.get(i), actualParts.get(i), String.format("result[%d]", i));
+        }
+    }
+
+    @Test
+    void splitCsvString_MissingValues() {
+        List<String> expectedParts = Arrays.asList("", "Smith", "", "");
+        String testString = String.join(",", expectedParts);
+
+        List<String> actualParts = SplitStringUtilities.splitCsvString(testString);
+
+        assertEquals(4, actualParts.size(), createVerificationMessage("result size"));
+
+        for(int i=0; i < actualParts.size(); i++) {
+            assertEquals(expectedParts.get(i), actualParts.get(i), String.format("result[%d]", i));
+        }
+    }
+
+    @Test
+    void splitCsvString_LeadingTrailingSpacesTrimmed() {
+        List<String> expectedParts = Arrays.asList("   ", " Smith  ");
```

Comment:
Same here.

Shouldn't these utility functions be "garbage in, garbage out", and trimming the the white space was going to be left to another function?